### PR TITLE
Add indexRandomForConcurrentSearch to tests

### PR DESCRIPTION
**Pull request by jed326**
Add indexRandomForConcurrentSearch to tests:
* CardinalityWithRequestBreakerIT.java
* TopHitsIT.java
* SearchFieldsIT.java
* DecayFunctionScoreIT.java
* FunctionScoreFieldValueIT.java
* FunctionScoreIT.java
* FunctionScorePluginIT.java
* QueryRescorerIT.java

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [x] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/internalClusterTest/java/org/opensearch/search/functionscore/FunctionScoreFieldValueIT.java [hunk 1]
```diff
@@ -99,7 +99,7 @@ public void testFieldValueFactor() throws IOException {
         client().prepareIndex("test").setId("1").setSource("test", 5, "body", "foo").get();
         client().prepareIndex("test").setId("2").setSource("test", 17, "body", "foo").get();
         client().prepareIndex("test").setId("3").setSource("body", "bar").get();
-
+        indexRandomForConcurrentSearch("test");
```
Comment (sohami) [inline server/src/internalClusterTest/java/org/opensearch/search/functionscore/FunctionScoreFieldValueIT.java:102]: should this be after `refresh()` ?  Such that actual documents are inserted in a segment and then these random docs get added to a new segment

File: server/src/internalClusterTest/java/org/opensearch/search/functionscore/FunctionScoreIT.java [hunk 2]
```diff
@@ -148,9 +149,10 @@ public void testScriptScoresNested() throws IOException {
         assertThat(response.getHits().getAt(0).getScore(), equalTo(1.0f));
     }
 
-    public void testScriptScoresWithAgg() throws IOException {
+    public void testScriptScoresWithAgg() throws IOException, InterruptedException {
         createIndex(INDEX);
         index(INDEX, TYPE, "1", jsonBuilder().startObject().field("dummy_field", 1).endObject());
+        indexRandomForConcurrentSearch(INDEX);
```
Comment (sohami) [inline server/src/internalClusterTest/java/org/opensearch/search/functionscore/FunctionScoreIT.java:155]: same here and other methods of this class

File: server/src/internalClusterTest/java/org/opensearch/search/functionscore/QueryRescorerIT.java [hunk 3]
```diff
@@ -831,6 +836,7 @@ public void testRescorePhaseWithInvalidSort() throws Exception {
         for (int i = 0; i < 5; i++) {
             client().prepareIndex("test").setId("" + i).setSource("number", 0).get();
         }
+        indexRandomForConcurrentSearch("test");
```
Comment (sohami) [inline server/src/internalClusterTest/java/org/opensearch/search/functionscore/QueryRescorerIT.java:839]: same here

File: server/src/internalClusterTest/java/org/opensearch/search/functionscore/FunctionScoreFieldValueIT.java (hunk 1 above)
Comment (jed326) [reply server/src/internalClusterTest/java/org/opensearch/search/functionscore/FunctionScoreFieldValueIT.java:102]: Yes they should, good catch. Let me fix all instances of this.

**Review by sohami — APPROVED** [review]

**Comment by sohami** [comment]
> ❕ Gradle check result for [27cb5cd](https://github.com/opensearch-project/OpenSearch/commit/27cb5cd1cbaa8d254f0db9b1c2d2686a9713c9c2): [UNSTABLE](https://build.ci.opensearch.org/job/gradle-check/30165/)
> 
> * **TEST FAILURES:**
> 
> ```
>       1 org.opensearch.cluster.allocation.AwarenessAllocationIT.testThreeZoneOneReplicaWithForceZoneValueAndLoadAwareness
> ```
> 
> Please review all [flaky tests](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) that succeeded after retry and create an issue if one does not already exist to track the flaky failure.

Existing issue: https://github.com/opensearch-project/OpenSearch/issues/3603

**Pull request merged**